destination/batches.go

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type streamNameParser interface {
+	ParseStreamName(sdk.Record) (string, error)
+}
+
+// fromColFieldParser parses the streamName from the opencdc.collection record
+// metadata field.
```

Comment:
I think this comment is more appropriate with [`ParseStreamName`](https://github.com/conduitio-labs/conduit-connector-kinesis/pull/51/files#diff-451bf27366f9e28315c11b17e205da29d4bd6dcb7302f668e93b87efc71e3346R33)